src/core/qgsfield.h

```diff
@@ -66,13 +66,15 @@ class CORE_EXPORT QgsField
      * @param prec Field precision. Usually decimal places but may also be
      * used in conjunction with other fields types (eg. variable character fields)
      * @param comment Comment for the field
+     * @param subType If the field is a collection, its element's type
      */
     QgsField( const QString& name = QString(),
               QVariant::Type type = QVariant::Invalid,
               const QString& typeName = QString(),
               int len = 0,
               int prec = 0,
-              const QString& comment = QString() );
+              const QString& comment = QString(),
+              QVariant::Type subType = QVariant::Invalid );
```

Comment:
@nyalldawson,
The widget to edit that would be way more complex (for the user and the programmer), I'd say.
This PR as it is now would not go in the way of implementing that. You can always leave subType to QVariant:Invalid...